src/Auth/Source/X509userCert.php

```diff
@@ -234,4 +260,33 @@ public function authSuccesful(array &$state): void
 
         throw new Exception("Should never be reached");
     }
+
+
+    /**
+     * Find user in LDAP-store
+     *
+     * @param string $attr
+     * @param string $value
+     * @return \Symfony\Component\Ldap\Entry|null
+     */
+    public function findUserByAttribute(string $attr, string $value): ?Entry
+    {
+        $searchBase = $this->ldapConfig->getString('search.base');
+
+        $searchUsername = $this->ldapConfig->getString('search.username');
+        Assert::notWhitespaceOnly($searchUsername);
+
+        $searchPassword = $this->ldapConfig->getOptionalString('search.password', null);
+        Assert::nullOrnotWhitespaceOnly($searchPassword);
+
+        $ldap = ConnectorFactory::fromAuthSource($this->backend);
+        $ldapUserProvider = new LdapUserProvider($ldap, $searchBase, $searchUsername, $searchPassword, [], $attr);
```

Comment:
Option search.base should be retrieved with getArray. That makes $searchBase an array of strings, while LdapUserProvider expects $searchBase to be a string.

A quick and dirty workaround, just for the sake of further testing, would be to pass $searchBase[0] (other base DNs to be searched are temporarily ignored).